src/D2L.CodeStyle.Analyzers/ApiUsage/DependencyInjection/DependencyRegistrationsAnalyzer.cs

```diff
@@ -208,13 +219,56 @@ ITypeSymbol type
 			return ImmutableArray.Create( registration.DependencyType );
 		}
 
-		private bool TryGetDependenciesFromConstructor( ITypeSymbol type, out ImmutableArray<ITypeSymbol> dependencies  ) {
-			var ctor = type.GetMembers()
+		private bool TryGetInstantiatedTypeForRegistration( DependencyRegistration registration, out ITypeSymbol instantiatedType ) {
+			if( registration.IsInstanceRegistration ) {
+				instantiatedType = null;
+				return false;
+			}
+
+			ITypeSymbol type = registration.FactoryType ?? registration.ConcreteType;
+
+			if( type.IsNullOrErrorType() ) {
+				instantiatedType = null;
+				return false;
+			}
+
+			// handle RegisterSubInterface<IFoo, IFooBar>():
+			// IFooBar is in the registration as "concrete type" and don't want to deal with that refactoring
+			if( type.TypeKind == TypeKind.Interface ) {

```

Comment:
The disconnect between the code that is looking for `RegisterSubInterface` and this code is pretty high.
Maybe we should revisit `DependencyRegistration`? Sounds like it isn't modelling our DI as good as it could be...

Reply:
For sure, it's probably enough to do one or the other with `Marker` (only the dependency type) and ignore the other. Or ignore the call all together? Don't know what analysis is done on it right now.